_inc/lib/core-api/wpcom-endpoints/id-range.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+class WPCOM_REST_API_V2_Endpoint_Id_Range {
+	public function __construct() {
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	public function register_routes() {
+		register_rest_route( 'wpcom/v2', '/id-range', array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_range' ),
+				'permission_callback' => array( $this, 'permission_callback' ),
+				'args'                => array(
+					'object_type' => array(
+						'type'    => 'string',
+						'default' => 'posts',
+						'enum'    => array(
+							'posts',
+							'comments',
+							'users'
+						)
+					),
+					'limit'       => array(
+						'type'    => 'integer',
+						'default' => 1000
+					),
+					'starting_id' => array(
+						'type'    => 'integer',
+						'default' => 0,
+					)
+				),
+			),
+		) );
+	}
+
+	public function permission_callback( $request ) {
+		return current_user_can( 'manage_posts' );
+	}
+
+	public function get_range( $request ) {
+		global $wpdb;
+
+		$type        = $request->get_param( 'object_type' );
+		$from        = $wpdb->{$type};
+		$limit       = $request->get_param( 'limit' );
+		$starting_id = $request->get_param( 'starting_id' );
+
+		$results = $wpdb->get_results(
```

Comment:
Rather than invoking SQL directly in this endpoint, we should operate via Sync modules.

Firstly, because the "object type" we intend to pass in here is actually a sync module name (which allows us to retrieve the module using `Jetpack_Sync_Modules::get_module( $module_slug )`, and secondly because sync modules contain functionality to (for example) filter out blacklisted post types when calculating which IDs to return.

Reply:
Also you should be using $wpdb->prepare here rather than injecting these params directly